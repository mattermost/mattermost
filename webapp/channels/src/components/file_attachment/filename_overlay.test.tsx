// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import FilenameOverlay from 'components/file_attachment/filename_overlay';
import AttachmentIcon from 'components/widgets/icons/attachment_icon';

describe('components/file_attachment/FilenameOverlay', () => {
    function emptyFunction() {} //eslint-disable-line no-empty-function
    const fileInfo = {
        id: 'thumbnail_id',
        name: 'test_filename',
        extension: 'jpg',
        width: 100,
        height: 80,
        has_preview_image: true,
        user_id: '',
        channel_id: 'channel_id',
        create_at: 0,
        update_at: 0,
        delete_at: 0,
        size: 100,
        mime_type: '',
        clientId: '',
        archived: false,
    };

    const baseProps = {
        fileInfo,
        handleImageClick: emptyFunction,
        compactDisplay: false,
        canDownload: true,
    };

    test('should match snapshot, standard display', () => {
        const wrapper = shallow(
            <FilenameOverlay {...baseProps}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, compact display', () => {
        const handleImageClick = jest.fn();
        const props = {...baseProps, compactDisplay: true, handleImageClick};
        const wrapper = shallow(
            <FilenameOverlay {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(AttachmentIcon).exists()).toBe(true);

        wrapper.find('a').first().simulate('click');
        expect(handleImageClick).toHaveBeenCalledTimes(1);
    });

    test('should match snapshot, with Download icon as children', () => {
        const props = {...baseProps, canDownload: true};
        const wrapper = shallow(
            <FilenameOverlay {...props}>
                <AttachmentIcon/>
            </FilenameOverlay>,
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(AttachmentIcon).exists()).toBe(true);
    });

    test('should match snapshot, standard but not downloadable', () => {
        const props = {...baseProps, canDownload: false};
        const wrapper = shallow(
            <FilenameOverlay {...props}>
                <AttachmentIcon/>
            </FilenameOverlay>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
