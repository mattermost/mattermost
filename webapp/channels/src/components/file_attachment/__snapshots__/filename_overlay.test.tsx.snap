// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/file_attachment/FilenameOverlay should match snapshot, compact display 1`] = `
<WithTooltip
  title="test_filename"
>
  <a
    className="post-image__name"
    href="#"
    onClick={[MockFunction]}
    rel="noopener noreferrer"
  >
    <AttachmentIcon
      className="icon"
    />
    test_filename
  </a>
</WithTooltip>
`;

exports[`components/file_attachment/FilenameOverlay should match snapshot, standard but not downloadable 1`] = `
<span
  className="post-image__name"
>
  test_filename
</span>
`;

exports[`components/file_attachment/FilenameOverlay should match snapshot, standard display 1`] = `
<div
  className="post-image__name"
>
  <WithTooltip
    title={
      Object {
        "defaultMessage": "Download",
        "id": "view_image_popover.download",
      }
    }
  >
    <ForwardRef
      aria-label="download"
      download="test_filename"
      href="/api/v4/files/thumbnail_id?download=1"
      location="filename_overlay"
    >
      test_filename
    </ForwardRef>
  </WithTooltip>
</div>
`;

exports[`components/file_attachment/FilenameOverlay should match snapshot, with Download icon as children 1`] = `
<div
  className="post-image__name"
>
  <WithTooltip
    title={
      Object {
        "defaultMessage": "Download",
        "id": "view_image_popover.download",
      }
    }
  >
    <ForwardRef
      aria-label="download"
      download="test_filename"
      href="/api/v4/files/thumbnail_id?download=1"
      location="filename_overlay"
    >
      <AttachmentIcon />
    </ForwardRef>
  </WithTooltip>
</div>
`;
